app/screens/convert_gm_to_channel/convert_gm_to_channel_form/convert_gm_to_channel_form.tsx

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {Text, View} from 'react-native';
+
+import {convertGroupMessageToPrivateChannel, switchToChannelById} from '@actions/remote/channel';
+import {isErrorWithMessage} from '@app/utils/errors';
+import {preventDoubleTap} from '@app/utils/tap';
+import Button from '@components/button';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+import {displayUsername} from '@utils/user';
+
+import {ChannelNameInput} from '../channel_name_input';
+import MessageBox from '../message_box/message_box';
+import {TeamSelector} from '../team_selector';
+
+import {NoCommonTeamForm} from './no_common_teams_form';
+
+const getStyleFromTheme = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            paddingVertical: 24,
+            paddingHorizontal: 20,
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 24,
+        },
+        errorMessage: {
+            color: theme.dndIndicator,
+        },
+    };
+});
+
+type Props = {
+    channelId: string;
+    commonTeams: Team[];
+    profiles: UserProfile[];
+    locale?: string;
+    teammateNameDisplay?: string;
+}
+
+export const ConvertGMToChannelForm = ({
+    channelId,
+    commonTeams,
+    profiles,
+    locale,
+    teammateNameDisplay,
+}: Props) => {
+    const theme = useTheme();
+    const styles = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+    const intl = useIntl();
+
+    const [selectedTeam, setSelectedTeam] = useState<Team>();
+    const [newChannelName, setNewChannelName] = useState<string>('');
+    const [errorMessage, setErrorMessage] = useState<string>('Something went wrong. Failed to convert Group Message to Private Channel');
+
+    const submitButtonEnabled = selectedTeam && newChannelName.trim();
+
+    const {formatMessage} = useIntl();
+    const confirmButtonText = formatMessage({
+        id: 'channel_info.convert_gm_to_channel.button_text',
+        defaultMessage: 'Convert to Private Channel',
+    });
+
+    useEffect(() => {
+        if (commonTeams.length > 0) {
+            setSelectedTeam(commonTeams[0]);
+        }
+    }, []);
+
+    const handleOnPress = useCallback(preventDoubleTap(async () => {
```

Comment:
Same out of scope problem with preventDoubleTap as I mentioned before. This one is less bad, since it won't reset the "has been pressed" state on every render, but still change the state when the callback dependencies change.

Reply:
Created https://mattermost.atlassian.net/browse/MM-55521 for double tap prevention refactor.